src/components/SubNav/SubNav.tsx

```diff
@@ -0,0 +1,379 @@
+      hasIconInChildren
+    );
+
+    return (
+      <Link
+        key={id}
+        id={id}
+        type={type}
+        href={href}
+        isUnderlined={false}
+        screenreaderOnlyText={screenreaderOnlyText}
+        __css={{
+          ...styles.a,
+          ...styles.outLine,
+          ...(isSelected ? styles.selectedItem : null),
+        }}
+      >
+        {content}
+      </Link>
+    );
+  }
+);
+
+export const SubNav: ChakraComponent<
+  React.ForwardRefExoticComponent<
+    React.PropsWithChildren<SubNavProps> & React.RefAttributes<HTMLDivElement>
+  >,
+  SubNavProps
+> = chakra(
+  forwardRef<HTMLDivElement, React.PropsWithChildren<SubNavProps>>((props) => {
+    const {
+      className,
+      actionBackgroundColor,
+      highlightColor,
+      selectedItem,
+      primaryActions,
+      secondaryActions,
+    } = props;
+
+    const [showRightFade, setShowRightFade] = useState(false);
+    const [lastScrollLeft, setLastScrollLeft] = useState(0);
+    const scrollableRef = useRef(null);
+
+    const styles = useMultiStyleConfig("SubNav", {
+      backgroundColor: actionBackgroundColor,
+      highlightColor: highlightColor,
+    });
+
+    if (actionBackgroundColor !== undefined && highlightColor === undefined) {
+      console.warn(
+        "NYPL Reservoir SubNav: The `actionBackgroundColor` prop has been passed, but the `highlightColor` prop has not been passed. Because of this, the `actionBackgroundColor` prop will be ignored."
+      );
+    }
+
+    const handleScroll = () => {
+      if (scrollableRef.current) {
+        const { scrollLeft, scrollWidth, clientWidth } = scrollableRef.current;
+
+        // Determine if we are scrolling to the left or right
+        const isScrollingRight = scrollLeft > lastScrollLeft;
+        setLastScrollLeft(scrollLeft); // Update last scroll position
+
+        // Show right fade only if scrolling to the left
+        if (!isScrollingRight) {
+          // Show right fade only if not at the end
+          const atRightEdge = scrollLeft + clientWidth >= scrollWidth;
+          setShowRightFade(!atRightEdge);
+        } else {
+          setShowRightFade(false); // Hide when scrolling right
+        }
+      }
+    };
+
+    useEffect(() => {
+      const refCurrent = scrollableRef.current;
+      if (refCurrent) {
+        refCurrent.addEventListener("scroll", handleScroll);
+        // Initial check to set the right fade effect
+        const { scrollWidth, clientWidth } = refCurrent;
+        setShowRightFade(scrollWidth > clientWidth); // Show fade if content is wider than the visible area
+      }
+
+      return () => {
+        if (refCurrent) {
+          refCurrent.removeEventListener("scroll", handleScroll);
+        }
+      };
+    }, []);
+
+    // Function to ensure only valid CSS properties are passed
+    const fadeEffectStyles = (styleObject) => {
+      const { accentColor, ...validStyles } = styleObject; // Exclude non-CSS properties
```

Comment:
I'm not sure I understand what `accentColor` is. It's not defined in `styles.fadeEffect` so this function doesn't change the input.